cmd/main_test.go

```diff
@@ -301,3 +315,59 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetUserNamespaces",
+	func(labelKey string, labelValues []string, expectedNamespaces []string) {
+		e := echo.New()
+
+		var req *labels.Requirement
+		var err error
+
+		// Create the label requirement based on the input
+		if len(labelValues) > 0 {
+			req, err = labels.NewRequirement(labelKey, selection.In, labelValues)
+		} else {
+			req, err = labels.NewRequirement(labelKey, selection.Exists, []string{})
+		}
+		Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+
+		namespaces, err := getUserNamespaces(e, *req)
+		Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+
+		var actualNamespaces []string
+		for _, ns := range namespaces {
+			actualNamespaces = append(actualNamespaces, ns.Name)
+		}
+
+		log.Printf("Expected Namespaces: %v, Actual Namespaces: %v", expectedNamespaces, actualNamespaces)
+
+		// Check if actual namespaces contain all expected namespaces
+		for _, expected := range expectedNamespaces {
+			Expect(actualNamespaces).To(ContainElement(expected))
+		}
+	},
+	Entry(
+		"retrieve all namespaces labeled as `user` namespaces",
+		"konflux.ci/type",
+		[]string{"user"},
+		[]string{"test-tenant", "test-tenant-2", "test-tenant-3"},
+	),
+	Entry(
+		"Get specific user namespace",
+		"kubernetes.io/metadata.name",
+		[]string{"test-tenant"},
+		[]string{"test-tenant"},
+	),
+	Entry(
+		"Get multiple specific user namespaces",
+		"kubernetes.io/metadata.name",
+		[]string{"test-tenant", "test-tenant-2"},
+		[]string{"test-tenant", "test-tenant-2"},
+	),
+	Entry(
+		"Get non-existent namespace/ checks to retrieve a namespace that doesn't exist",
+		"kubernetes.io/metadata.name",
+		[]string{"non-existent-namespace"},
+		[]string{},
+	),
+)
```

Comment:
Why did we create `createNamespaceWithoutLabels` if we don't have a testcase for it?